tests/unit/src/unit/issues/Issue7903.hx

```diff
@@ -5,14 +5,29 @@ import utest.Assert;
 class Issue7903 extends unit.Test {
 	static var tmp:String;
 	function test() {
+		//Test recursive objects
 		var o = {rec:(null:Dynamic)};
 		o.rec = o;
 		try {
 			tmp = Std.string(o);
 			Assert.pass();
 		} catch(e:Dynamic) {
-			trace(e);
-			Assert.fail('Failed to stringify an object with recursive reference');
+			Assert.fail('Failed to stringify an object with recursive reference: $e');
+		}
+
+		//Test recursive enums
+		var o = {rec:(null:RecursiveEnum)};
+		var e = Ctor(o);
+		o.rec = e;
+		try {
+			tmp = Std.string(e);
+			Assert.pass();
+		} catch(e:Dynamic) {
+			Assert.fail('Failed to stringify an enum instance with recursive reference: $e');
 		}
 	}
+}
+
+private enum RecursiveEnum {
+	Ctor(o:{rec:Null<RecursiveEnum>});
```

Comment:
It would be better to test this without a structure, else the structure recursion check could handle it while it still fails for pure enums.

Reply:
How do I make an enum ctor referencing itself if it's immutable? :)
It's either through an object or an array.

Reply:
Right, I guess that doesn't make sense...

Speaking of arrays though... :)